design/threat-model/threat-model.md

```diff
@@ -0,0 +1,138 @@
+Virtio-net: lalala, unless you use direct device assignment, blah blah
+Virtio-vsock: In order to interact with the guest agent and retrieve STDIO from containers, a medium of communication is required. This is available via virtio-vsock.
+
+Other interfaces which may be utilized include VFIO (direct device assignment) and ACPI for dynamic VM resource management (ie, hotplug).
+
+How these devices are utilized varies depending on the VMM utilized. We clarify the default settings provided when integrating Kata with the QEMU VMM, Firecracker VMM, and Cloud Hypervisor VMM (integration of Cloud Hypervisor is in progress, expected in 1.10 release of Kata Containers) in the following sections.
+
+### Devices
+
+#### virtio-blk/-scsi
+
+The backend for virtio-blk and -scsi are based in the VMM itself (ring3 in the context of x86) by default.
+While vhost based backends are available for QEMU, it is not recommended. vhost-user backends are under consideration
+in the future, but not utilized in any of the existing Kata configurations.
+
+#### virtio-FS
+
+virtio-FS's interaction with the host filesystem is done through a vhost-user daemon. The virtio-fs client, running in
+the guest, will generate requests to access files. `virtiofsd` will receive requests, open the file, and request the VMM to `mmap` it into the guest. The guest will access the file using DAX, utilizing the host's page cache, thus avoiding copy and duplication.
+
+This is still new development. DAX-less support is available in the 5.4 Linux kernel. QEMU VMM supports virtio-fs as of v4.2,
+and upstreaming of `virtiofsd` is in progress.
+
+#### virtio-net
+
+Virtio-net has many options, depending on the VMM and Kata configurations.
+
+##### QEMU
+
+While QEMU has options for vhost, virtio-net and vhost-user, the virtio-net backend
+for Kata defaults to vhost-net for performance reasons. The default configuration is being
+reevaluated.
+
+##### Firecracker
+
+For Firecracker, the backend is within Firecracker's VMM.
+
+##### Cloud Hypervisor
+
+For Cloud Hypervisor, the current backend default is virtio-net (in the VMM). vhost-user-net support
+is being added (written in rust, Cloud Hypervisor specific).
+
+#### virtio-vsock
+
+##### QEMU
+
+In QEMU, vosck is backed by `vhost_vsock`, which runs within the kernel itself (ring0 in the context of x86).
+
+##### Firecracker and Cloud Hypervisor
+
+In Firecracker and Cloud Hypervisor, vsock is backed by a unix-domain-socket in the hosts userspace.
+
+#### VFIO
+
+Utilizing VFIO, devices can be passed through to the virtual machine. We will assess this separately. Exposure to
+host is limited to gaps in device pass-through handling. This is supported in QEMU and Cloud Hypervisor, but not
+Firecracker.
+
+
+#### ACPI
+
+ACPI is necessary for hotplug of CPU, memory and devices. ACPI is available only in QEMU. In Firecracker and 0.3 Cloud Hypervisor,
+there isn’t the capability to hotplug deviecs, memory, or CPUs.
```

Comment:
/deviecs/devices

Reply:
Maybe worth explaining why hotplugging is so important for Kata? (that was not obvious to me initially)

Reply:
Let's add that in more clarity into our arch docs